src/supervisor/watchers/handlers/index.ts

```diff
@@ -292,12 +430,13 @@ export async function setupInformer(
   }
 
   const workloadMetadata = workloadWatchMetadata[workloadKind];
-  const namespacedEndpoint = workloadMetadata.namespacedEndpoint.replace(
-    '{namespace}',
-    namespace,
-  );
+  const endpoint = shouldWatchCluster
+    ? workloadMetadata.clusterEndpoint
+    : workloadMetadata.namespacedEndpoint.replace('{namespace}', namespace);
 
-  const listMethod = workloadMetadata.namespacedListFactory(namespace);
+  const listMethod = shouldWatchCluster
+    ? workloadMetadata.clusterListFactory()
+    : workloadMetadata.namespacedListFactory(namespace);
```

Comment:
Do we want to know whether the error comes from namespacedListFactory or clusterListFactory?